portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java

```diff
@@ -51,4 +53,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.getWriter().println(gson.toJson(userReviews));
     }
   }
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    
+    String userReviewText = request.getParameter("reviewText");
+    String isbn = request.getParameter("isbn");
+    String email = userService.getCurrentUser().getEmail();
+    
+    Review userReview = Review.create(userReviewText,isbn,email);
+    try {
+        reviewDao.uploadNew(userReview);
+    } catch (ServletException ex) {
```

Comment:
nit: this middle catch block for ServletException might be a little redundant since it does the exact same as the Exception catch block. 

ServletException class inherits from Exception (https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/ServletException.html), so I would think the 'Exception' block would still catch ServletExceptions.

Reply:
I might have confused you, I think we want to throw the servlet exception here.
``` 
   try {
      reviewDao.uploadNew(userReview);
    } catch (Exception ex) {
      throw new ServletException("Failed to save review, please ensure review doesn't already exists for this book", ex);
    }
```